cpp/collective/rendezvous/src/tcp_store.cpp

```diff
@@ -1313,4 +1314,16 @@ void TCPStore::multiSet(const std::vector<std::string> &keys,
 
 bool TCPStore::hasExtendedApi() const { return true; }
 
+void TCPStore::set(const std::string &key, const std::vector<char> &data) {
+    std::vector<uint8_t> dataSet(data.begin(), data.end());
```

Comment:
Here copying data to cast the `vector<char>` to `vector<uint8_t>`. But the send socket accepts the `const char *` type here https://github.com/xorbitsai/xoscar/blob/main/cpp/collective/rendezvous/include/utils.hpp#L121:
```c
::send(socket, (const char *) currentBytes, bytesToSend, flags)
```

We should unify the data types to char* to avoid copying data.

Reply:
> Here copying data to cast the `vector<char>` to `vector<uint8_t>`. But the send function accepts a `const char *` type here https://github.com/xorbitsai/xoscar/blob/main/cpp/collective/rendezvous/include/utils.hpp#L121:
> 
> ```c
> ::send(socket, (const char *) currentBytes, bytesToSend, flags)
> ```
> 
> We should unify the data types to char* to avoid copying data.

See issue https://github.com/pybind/pybind11/issues/1807 in pybind11. Copying data here is just for pybind11.